### PR TITLE
Add optional metadatacompliancetype field to object-templates

### PR DESCRIPTION
**Pull request by willkutler**
https://github.com/open-cluster-management/backlog/issues/17904

Signed-off-by: Will Kutler <wkutler@redhat.com>

### CODE REVIEW

**Review by ycao56 — APPROVED** [review]

**Comment by gparvin** [comment]
Is there a good use case for:
```
    - complianceType: musthave
      metadataComplianceType: mustonlyhave
```

@willkutler  Can you link a doc pr to the issue if this will be documented?  Otherwise we'll probably forget to document this field.

**Comment by willkutler** [comment]
@gparvin a use case for that would be if a user wants to use mustonlyhave to remove an existing label or annotation without changing the behavior of the rest of the policy. Should I be making a doc PR myself or just creating an issue for it?

**Comment by gparvin** [comment]
@willkutler Just a doc issue I meant to say.  Thanks!

**Review by gparvin — APPROVED** [review]

File: test/e2e/case9_md_check_test.go [hunk 1]
```diff
@@ -136,4 +142,45 @@ var _ = Describe("Test pod obj template handling", func() {
 			}, defaultTimeoutSeconds, 1).Should(Equal("Compliant"))
 		})
 	})
+	Describe("Create a namespace policy on managed cluster in ns:"+testNamespace, func() {
+		It("should create a namespace with multiple annotations on the managed cluster", func() {
+			By("Creating " + case9MultiAnnoNSCreate + " on managed")
+			utils.Kubectl("apply", "-f", case9PolicyYamlMultiAnnoNSCreate, "-n", testNamespace)
```
Comment (mprahl) [inline test/e2e/case9_md_check_test.go:148]: Where does this policy get cleaned up?

**Comment by willkutler** [comment]
@mprahl added some changes based on your comments

**Review by mprahl — APPROVED** [review]
Nice!

**Review by mprahl — CHANGES_REQUESTED** [review]

**Review by mprahl — APPROVED** [review]
Nice!

**Comment by willkutler** [comment]
Doc issue for this here: https://github.com/open-cluster-management/backlog/issues/18649

**Pull request merged**